File: app/src/main/java/com/jeeldobariya/passcodes/utils/Permissions.java
package com.jeeldobariya.passcodes.utils;

import android.Manifest;
import android.app.Activity;
import android.content.pm.PackageManager;
import androidx.annotation.NonNull;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

public class Permissions {

  public static final int PERMISSION_REQUEST_CODE = 100;
  private final Activity activity;

  public Permissions(Activity activity) {
    this.activity = activity;
  }

  // Check if permission has been granted
  public boolean checkPermission() {
    int resultWrite = ContextCompat.checkSelfPermission(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE);
    int resultRead = ContextCompat.checkSelfPermission(activity, Manifest.permission.READ_EXTERNAL_STORAGE);
    return resultWrite == PackageManager.PERMISSION_GRANTED && resultRead == PackageManager.PERMISSION_GRANTED;
  }

  // Request write and read file permissions
  public void requestPermission() {
    ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE}, PERMISSION_REQUEST_CODE);
  }

  // Check if permission has been granted after the request
  public boolean isPermissionGranted(@NonNull int[] grantResults) {
    return grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED && grantResults[1] == PackageManager.PERMISSION_GRANTED;
  }
}

File: app/src/main/kotlin/com/jeeldobariya/passcodes/utils/Permissions.kt
package com.jeeldobariya.passcodes.utils

import android.Manifest
import android.app.Activity
import android.content.pm.PackageManager
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat

class Permissions(private val activity: Activity) {

    companion object {
        const val PERMISSION_REQUEST_CODE = 100
    }

    /**
     * Check if write and read external storage permissions have been granted.
     * @return True if permissions are granted, false otherwise.
     */
    fun checkPermission(): Boolean {
        val resultWrite = ContextCompat.checkSelfPermission(activity, Manifest.permission.WRITE_EXTERNAL_STORAGE)
        val resultRead = ContextCompat.checkSelfPermission(activity, Manifest.permission.READ_EXTERNAL_STORAGE)
        return resultWrite == PackageManager.PERMISSION_GRANTED && resultRead == PackageManager.PERMISSION_GRANTED
    }

    /**
     * Request write and read external storage permissions from the user.
     */
    fun requestPermission() {
        ActivityCompat.requestPermissions(
            activity,
            arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE),
            PERMISSION_REQUEST_CODE
        )
    }

    /**
     * Check if permissions were granted after a request.
     * This method is typically called in [Activity.onRequestPermissionsResult].
     * @param grantResults The results for the requested permissions.
     * @return True if both requested permissions were granted, false otherwise.
     */
    fun isPermissionGranted(grantResults: IntArray): Boolean {
        // We expect two permissions (WRITE_EXTERNAL_STORAGE, READ_EXTERNAL_STORAGE)
        // and check if both were granted.
        return grantResults.isNotEmpty() &&
               grantResults.size >= 2 && // Ensure we got results for at least two permissions
               grantResults[0] == PackageManager.PERMISSION_GRANTED &&
               grantResults[1] == PackageManager.PERMISSION_GRANTED
    }
}
